lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -514,7 +514,7 @@ public function getAlterTableSQL(TableDiff $diff)
                     );
                 } elseif ($hasFromComment && ! $hasComment) {
                     $commentsSql[] = $this->getDropColumnCommentSQL($diff->name, $column->getQuotedName($this));
-                } elseif ($hasComment) {
+                } elseif (! $hasFromComment && $hasComment) {
```

Comment:
Can you inverse these conditionals? They read very funny with the `!` in front

Reply:
Fixed.